### DISCUSSION TITLE
Question about ERA-201 fuse

### DISCUSSION THREAD

**Discussion by AlexBuno** [lpollier/battmon#5] [Q&A] [answered]
Hi! Thanks for the work you have done, I have two ers210 batteries that I am rebuilding. But one of them has a dead fuse, and I can't seem to find it anywhere to buy, do you have any advice? Also, is it necessary to reprogram the EEPROM? Won't the battery recalibrate to the new cell capacity?

Thanks
Alex

**Comment by lpollier** [comment, marked as answer]
Hello Alex,

Perhaps you were able to use the reverse engineering schematic of the ERA-201B1 battery available here:
https://github.com/lpollier/battmon/blob/master/example/ERA-201B1_reverse_engineering_schematic.pdf

In it, you can find the part number of the main components, or references that I think are closest to the origin.
For example, the fuse F1 is assigned the reference SFD-047A which is closer to the original component, and is available at Aliexpress in batches of 3 or 10 currently. I personally used 2 of them to replace the fried fuses.

About the EEPROM content after re-cell, it would seem that the gas gauge bq2040 is able to "relearn" the real capacity of new cells after performing one or more complete charge/discharge cycles, as indicated in the datasheet and summarized in the tutorial.
But I haven't tried it, so I'm relying on several testimonials in one of the robotcenter discord chats.

Hope that help.
Loic

**Comment by AlexBuno** [comment]
Hi Loic

Thanks for your quick reply, yes I did check the schematic (I am an
electrical engineer, so I really appreciated the reverse engineering you
did)
I saw the SFD-047A part number, and the seller on Aliexpress, but the
Aliexpress seller seemed a bit fishy (They don't mention the fuse breaking
capacity), and I was trying to find an equivalent I could buy in Europe.

I'm also curious about chapter 5 in your battery recelling manual (5.
Solder the pack to the board and reinitialize the battery protection) You
mention  "reinitialize the reference of the first level protection circuit
IC1" and "However, if you do it directly, you will probably destroy fuse F1
via transistor Q8." But I don't see how IC1 relates to the fuse based on
your schematic. It looks like only IC21 has control over blowing the fuse
through Q8, and IC21 will only activate Q8 if the voltage difference
between Vbat+ and Vbat - is 4.35x2 so 8.7v? So I don't see why the fuse
would blow when connecting the board to the new cells? Perhaps there is a
voltage spike when connecting?

Additionally, have you tried replacing the potentiometers in the ERS210
limbs? I couldn't find any information online, but I think these are the
potentiometers (
https://lcsc.com/product-detail/Position-Sensor_ALPSALPINE-RDC503051A_C470764.html
)

Are you working on any other robot repairs? I'm doing a PhD on business
models for repair, and I love to repair, and I'm working to increase
the repair rate of devices. I also repair robotic devices in my spare time.

Thanks
Alex Bunodiere

On Mon, Nov 28, 2022 at 9:19 AM Loïc POLLIER ***@***.***>
wrote:

> Hello Alex,
>
> Perhaps you were able to use the reverse engineering schematic of the
> ERA-201B1 battery available here:
>
> https://github.com/lpollier/battmon/blob/master/example/ERA-201B1_reverse_engineering_schematic.pdf
>
> In it, you can find the part number of the main components, or references
> that I think are closest to the origin.
> For example, the fuse F1 is assigned the reference SFD-047A which is
> closer to the original component, and is available at Aliexpress in batches
> of 3 or 10 currently. I personally used 2 of them to replace the fried
> fuses.
>
> About the EEPROM content after re-cell, it would seem that the gas gauge
> bq2040 is able to "relearn" the real capacity of new cells after performing
> one or more complete charge/discharge cycles, as indicated in the datasheet
> and summarized in the tutorial.
> But I haven't tried it, so I'm relying on several testimonials in one of
> the robotcenter discord chats.
>
> Hope that help.
> Loic
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lpollier/battmon/discussions/5#discussioncomment-4251084>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXUVIYXDRWJWQQA65MYKJOTWKRTI3ANCNFSM6AAAAAASMYGIVA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Regards
Alex Bunodiere

**Comment by lpollier** [comment]
There is an indirect link between the first level battery protection component IC1 and the second level IC21.

When you have resoldered the new cells, you must then reinitiate IC1, so that BAT- and PACK- find the same potential for a short time. This step eventually causes an overvoltage detected by IC21 and then activates transistor Q8, which shorts fuse F1.

To avoid this trap, it is proposed in the tutorial to inhibit the transistor Q8 the time to carry out the manipulation concerning the reinitialization of IC1.

Furthermore, it is also possible to cause this overvoltage detection at the moment of soldering the new cells.
So this 2-in-1 technique of inhibiting Q8 avoids both the potential overvoltage when soldering the cells and when resetting IC1.

**Comment by lpollier** [comment]
About the fuse you can get the official datasheet here:
https://github.com/lpollier/battmon/blob/master/example/ERA-201B1_kicad/libraries/SFD-047A/SFD-047A.pdf

Since this is a deprecated component, and it looks there is no perfect equivalent component on production, it is therefore necessary to obtain ends of reels from the retailer of your choice. Or find an almost equivalent in terms of electrical characteristics and PCB footprint.

Also, I have no idea what the leg potentiometer part number is because I haven't encountered this problem yet. There is some auctions about them on Zenmarket for example to compare your findings.

**Comment by AlexBuno** [comment]
Hi Loic

I see.. i guess another option would be to remove the solder ball on BL2,
and that will prevent the transistor from shorting the fuse?

Thanks for your help. For your information I think this is the correct
potentiometer, just missing an apparently custom plastic ear for extra
stability (https://www.lcsc.com/product-detail/_ALPSALPINE-_C470764.html)

Regards
Alex

On Mon, Nov 28, 2022 at 6:53 PM Loïc POLLIER ***@***.***>
wrote:

> There is an indirect link between the first level battery protection
> component IC1 and the second level IC21.
>
> When you have resoldered the new cells, you must then reinitiate IC1, so
> that BAT- and PACK- find the same potential for a short time. This step
> eventually causes an overvoltage detected by IC21 and then activates
> transistor Q8, which shorts fuse F1.
>
> To avoid this trap, it is proposed in the tutorial to inhibit the
> transistor Q8 the time to carry out the manipulation concerning the reset
> of IC1.
>
> Furthermore, it is also possible to cause this overvoltage detection at
> the moment of soldering the new cells.
> So this 2-in-1 technique of inhibiting Q8 avoids both the potential
> overvoltage when soldering the cells and when resetting IC1.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lpollier/battmon/discussions/5#discussioncomment-4255902>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXUVIYSZHEEVJTEXKTW7HMTWKTWRPANCNFSM6AAAAAASMYGIVA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Regards
Alex Bunodiere

**Comment by lpollier** [comment]
I saw your discussions regarding the leg potentiometer on the discord. That is a good information, thanks.

About BL2, I didn't try that way. I'm afraid that the PNP transistor Q7 will be triggered by the NMOS Q8, which will enter the continuous activation of it. And when resoldering BL2, the short circuit of F1 is made.

If it ever works without blowing the fuse, don't hesitate to come back and say it here.

**Comment by AlexBuno** [comment]
Hi Loic

I finally got around to continuing work on my AIbos :) I expanded my collection, and have 7xERS210 now and 1xERS-7. I made good progress on cleaning all the leg and tail pots of all the ERS210s as most had jitters. Now I'm continuing with battery recalling. I have already done two, thanks to your great guide. I also made the Battmonn, and used it on my batteries. I just had a quick question. Regarding this line in your guide "To conclude, it will be useful to perform a full charge and then a full discharge, so that the bq2040 gas gauge IC41 determines the exact capacity of the battery. This calibration can be done either via the charger in discharge mode or with the BattMon 2.0 toolkit.." when you say the calibration can be done via the charger in discharge mode, what are you referring to? Also, can I link your work to a full refurbishment guide I am making for the ERS210? I am busy gathering all repair info for the ERS210 and put it in one document, with all links, videos, pictures etc, together with basic steps to follow, and referring to the guides for complete steps. Thanks, Alex

**Reply by lpollier** [reply]
Hi Alex,

Congratulations on your progress with all the AIBO repairs and rehabilitations!

Regarding your questions:
1. Calibration can be achieved by charging the battery normally, followed by discharging it. You have a couple of options for discharging: you can use the discharge mode of the HTRC IMAX B6, which allows you to control the discharge current, or you can connect any load, such as the toolkit BattMon 2.0 or your AIBO.
-> The crucial point after the full charge is that the battery need to reach the low voltage level between 6.3V to 6.0V but not lower, in order to ensure that the bq2040 can update the EEPROM memory with the first measured capacity. Note that this is not obligatory, but it allows the battery to be charged/discharged over its entire effective voltage range.
2. You have complete freedom to utilize all code and documents under the terms of the MIT license. Feel free to link to them in any way you see fit.

Best,
Loic

**Comment by AlexBuno** [comment]
Hi Loic

Thanks! I'm making steady progress, currently, I'm working on the
batteries and also trying to find a solution for the slipping head
clutches. Considering glueing, replacing clutch material or unibody clutch
(either 3D printed or CNC machined)

I had a question about the Battmonn, in your guide you recommend
resetting the cycle count and adjusting the design capacity of the battery
by connecting to the EEPROM and changing the values. But as far as I can
tell you should be able to do the same thing with the Battmonn by writing a
word? (see below)

[image: image.png]

The thing is, I tried this. I wrote addresses 0x0E and 0x0F as 0x00, and I
got the "OK" message, but the values at the addresses remained the same as
they were. Any idea why this may be? Is the EEPROM read-only?

Thanks
Alex

On Mon, May 6, 2024 at 9:09 PM Loïc POLLIER ***@***.***>
wrote:

> Hi Alex,
>
> Congratulations on your progress with all the AIBO repairs and
> rehabilitations!
>
> Regarding your questions:
>
>    1. Calibration can be achieved by charging the battery normally,
>    followed by discharging it. You have a couple of options for discharging:
>    you can use the discharge mode of the HTRC IMAX B6, which allows you to
>    control the discharge current, or you can connect any load, such as the
>    toolkit BattMon 2.0. The crucial point is that the battery needs to reach
>    the low voltage level of 6.0V to 6.3V.
>    2.You have complete freedom to utilize all code and documents under
>    the terms of the MIT license. Feel free to link to them in any way you see
>    fit.
>
> Best,
> Loic
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lpollier/battmon/discussions/5#discussioncomment-9333391>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXUVIYS67OSBSS56TJD7U4DZA7INPAVCNFSM6AAAAAASMYGIVCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TGMZTGM4TC>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Regards
Alex Bunodiere

**Reply by lpollier** [reply]
1. Regarding the problem of the clutch spinning endlessly, I solved it by using a drill press to create a hole in the shaft then inserting a metal rod with a little glue, as suggested in the Robotcenter Discord, but you have to be meticulous and it is quite a time-consuming process.

2. Concerning resetting the Cycle Count and adjusting the Design Capacity and Full Charge Capacity values, unfortunately, the BattMon 2.0 toolkit is limited to communicating with the bq2040 gas gauge and cannot adjust the values saved in the 24LC01 EEPROM memory. While it is possible to read and occasionally write to the bq2040 registers, this capability is generally not very useful and does not aid in re-celling the battery.
-> You'll need an I2C EEPROM memory programmer, like the Bus Pirate I suggested, or a similar tool, for this step. However, it seems this may not be necessary if you do the final calibration, as the gas gauge should adapt and store the new capacity automatically. I can't confirm this because I reset the memory on the four batteries I have.

Cheers,
Loic
